types.bal

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License. 
+
+# Represents configuration parameters to create Azure Cosmos DB client.
+# 
+# + baseUrl - Base URL of the Azure Cosmos DB account
+# + masterOrResourceToken - The token used to make the request call authorized
+public type Configuration record {|
+    string baseUrl;
+    string masterOrResourceToken;
+|};
+
+# Represents matadata headers which will return for a delete request.
+# 
+# + sessionToken - Session token from the response
+public type DeleteResponse record {|
+    string sessionToken;
+|};
+
+# Represents the common elements representing information.
+# 
+# + resourceId - A unique identifier which is used internally for placement and navigation of the resource
+# + selfReference - A unique addressable URI for the resource
+# + eTag - Resource etag for the resource retrieved
+# + sessionToken - Session token of the request
+public type Commons record {|
+    string resourceId?;
+    string selfReference?;
+    string eTag?;
+    string sessionToken?;
+|};
+
+# Represents the elements representing information about a database.
+# 
+# + id - User generated unique ID for the database 
+public type Database record {|
+    string id;
+    *Commons;
+|};
+
+# Represents the elements representing information about a container.
+# 
+# + id - User generated unique ID for the container
+# + indexingPolicy - Record of type IndexingPolicy
+# + partitionKey - Record of type PartitionKey
+public type Container record {|
+    string id;
+    *Commons;
+    IndexingPolicy indexingPolicy;
+    PartitionKey partitionKey;
+|};
+
+# Represent the parameters representing information about a document.
+# 
+# + id - User generated unique ID for the document 
+# + documentBody - The document reprsented as a map of json
+public type Document record {|
+    string id;
+    *Commons;
+    map<json> documentBody;
+|};
+
+# Represent the parameters necessary to create an indexing policy when creating a container.
+# 
+# + indexingMode - Mode of indexing
+# + automatic - Specify whether indexing is done automatically or not
+# + includedPaths - Array of type IncludedPath representing included paths
+# + excludedPaths - Array of type IncludedPath representing excluded paths
+public type IndexingPolicy record {|
+    string indexingMode;
```

Comment:
Shouldn't this be an `enum` with finite set of type (instead of a `string`)?

Reply:
Provided an enum type for indexingMode